src/praisonai-agents/tests/test_self_reflection_fix_simple.py

```
@@ -0,0 +1,37 @@
+
+    # Test with a simple calculation that might trigger self-reflection
+    try:
+        response = agent.start("What is 25 * 17? Show your work and double-check the answer.")
+        print(f"\nResponse: {response}")
+        
+        assert response, "Self-reflection with tools failed to produce a response."
+        print("\n✅ SUCCESS: Self-reflection with tools is working!")
+        
+    except Exception as e:
+        print(f"\n❌ ERROR: {str(e)}")
+        raise AssertionError(f"Test failed with error: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)